service/frontend/workflowHandler.go

```diff
@@ -3579,19 +3702,18 @@ func (wh *WorkflowHandler) validateTaskList(t *types.TaskList, scope metrics.Sco
 	if t == nil || t.GetName() == "" {
 		return wh.error(errTaskListNotSet, scope)
 	}
-	if !wh.validIDLength(t.GetName(), scope, domain) {
+
+	if !common.ValidIDLength(
+		t.GetName(),
+		scope,
+		wh.config.MaxIDLengthWarnLimit(),
+		wh.config.TaskListNameMaxLength(domain),
+		metrics.CadenceErrTaskListNameExceededWarnLimit) {
 		return wh.error(errTaskListTooLong, scope)
 	}
-	if wh.isRawListList(t) && len(t.GetName()) > wh.config.MaxRawTaskListNameLimit(domain) {
```

Comment:
For scalable tasklist I think we will generate task list names based on the value user provided, and the generated name may exceed the limit. Will that be a concern and stop it from working? I guess it's probably fine as those checks is only enforced frontend and decision where only raw task list name is used. Not sure if there's a case I miss.

Reply:
Yes I think its fine. Basically these checks will operate on the raw task list and we just need to make sure we reduce this limit to a point where we have enough buffer to tolerate the extension of scalable task list.